hack/update-vendor.sh

```diff
@@ -23,4 +23,13 @@ source "${KUBE_ROOT}/hack/ensure-go.sh"
 
 go mod tidy
 go mod vendor
+
+# Copy full dependencies if needed.
+for dep in $(cat ${KUBE_ROOT}/go.vendor); do
+    src=$(go mod download -json ${dep} | jq -r .Dir)
+    dst="${KUBE_ROOT}/vendor/${dep}"
```

Comment:
I needed a `mkdir -p vendor/${dep}`

Reply:
Is this to vendor a dependency you don't actually import?

Reply:
ah this was trying out the commands by hand, i think the assumption with this code is that vendor will always exist already, so it's probably fine to ignore, but couldn't hurt to add a mkdir -p $dst. i'm fine omitting.

Reply:
Right before this part there is `go mod vendor` which should take care of it, I had the mkdir -p before then I removed it to avoid adding things to vendor that aren't imported explicitly